ouroboros-consensus/src/consensus-testlib/Test/Ouroboros/Consensus/ChainGenerator/Adversarial.hs

```diff
@@ -73,29 +73,29 @@ data AdversarialViolation hon adv =
     BadRace !(RaceViolation hon adv)
   deriving (Eq, Read, Show)
 
--- | Check the chain matches the given 'AdversarialRecipe', especially the Extended Praos Race Assumption
+-- | Check the chain matches the given 'AdversarialRecipe', especially that it
+-- satisfies Adversarial Chains in Stability Windows and Length of Competing
+-- Chains.
 --
--- Definition of a /Praos Race Window/. Such a window includes exactly 'Delta'
--- slots after the @k+1@st honest block in the window.
+-- Definition of a /Praos Race Window/ of a chain. It is an interval of slots
+-- that contains at least @k+1@ blocks of the chain and exactly 'Delta' slots
+-- after the @k+1@st block.
 --
--- Definition of the /Praos Race Assumption/. We assume every adversarial chain
--- contains at most @k@ blocks in the Praos Race Window anchored at the
--- intersection.
+-- Definition of the /Length of Competing Chains/. We assume every adversarial
+-- chain contains at most @k@ blocks in the Praos Race Window starting at the
+-- next slot after the intersection.
 --
--- Definition of the /Extended Praos Race Assumption/. Beyond the Praos Race
--- Assumption, we assume that every adversarial chain also has at most @k@
--- blocks in every subsequent Praos Race Window that ends within 'Delta' of the
--- Stability Window anchored at the first adverarial block. (Crucially, these
--- are the Praos Race Windows in which the the adversary could not yet have
+-- Definition of the /Adversarial Chains in Stability Windows/.
+-- Every adversarial chain has at most @k@ blocks in every Praos Race Window of
+-- the honest chain and ending within 'Delta' blocks of the Stability Window
+-- starting at the slot of the first adversarial block. (Crucially, these
```

Comment:
I think there's a typo in the current text. Is the following suggestion accurate?

```suggestion
-- Every adversarial chain has at most @k@ blocks in every Praos Race Window
-- that ends at least 'Delta' slots before the Stability Window
-- anchored at the first adversarial block. (Crucially, these
```

Reply:
I edited your suggestion as I think it would make it accurate. I don't think anchoring will make it easy to express the same stability window.

The definition I have of anchored window is:
> A window anchored at a block starts with the slot immediately after that block.

... from `Honest.hs`.

Reply:
491a9d9 corrects the substitution of blocks with slots.